src/server/Agent.js

```diff
@@ -31,6 +31,11 @@ export default class Agent extends SideAgent {
         this._actualListeningAddress = null; // used (and fully resolved)
 
         this._transaction = new Transaction(this);
+
+        this._serverClosed = null; // a promise to close this.server
+
+        // the first accepted or inherited connection
+        this._firstConnectionSocket = null;
```

Comment:
Yeah. In other words, we rely on transaction.run() to at least start the transaction, which is a reasonable invariant. We also interpret started() as "we have begun to start" rather than "we have completed all the starting procedures" OR we assume that start is synchronous. That interpretation/assumption matches the current code, but the name of that started() method does not relay those nuanced well, unfortunately.